### PR TITLE
Add conv per-tensor input zero point.

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:

- Enable per-tensor zp on AVX512_VNNI and AVX512_AMX ISA to use BRGCONV kernel.
- Merge the @luo-cheng2021 fix PR https://github.com/openvinotoolkit/openvino/pull/12877
- Merge optimization of convolution attribute PR to lower down the load network time latency.https://github.com/openvinotoolkit/openvino/pull/12898
- Only enable conv/group conv S8 input on AMX. Non-AMX same with 2022.1 LPT input precision config. 

### Tickets:
 - *88228, 91221,91868*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky may I ask you to review the PR?

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:941]: Why those ones? We need to leave a comment

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:933]: Please use empty() / !empty() instead.

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:986]: Didn't get the idea of using array.
It seems it would be more clean to just use a vector, then insert attrs in scope of SetPostOpsAndZeroPoints
Then attrsNum variable is seemed to be unnecessary.

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:1054]: Also please use empty() instead of == 0

**Comment by EgorDuplensky** [comment]
@luweizhou2016 Could you please also write some description in PR message

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:999]: Please also leave a comment for what this iteration over different attrs is about.

File: src/plugins/intel_cpu/src/graph_optimizer.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/graph_optimizer.cpp:629]: Please leave a comment why we are having different zero points here.

**Review by EgorDuplensky — COMMENTED** [review]
In general I think we need to comment such workarounds as much as possible, especially when it is impossible to isolate them.

**Comment by luweizhou2016** [comment]
@EgorDuplensky Review comments applied. Thx!

File: src/plugins/intel_cpu/src/graph_optimizer.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/graph_optimizer.cpp:630]: I meant a bit different comment.
As far as I understand we prepare two types of zeropoints here and 
we let the node to decide later, which one should be actually used.
Can we have the similar explanation in the comment?

**Comment by luweizhou2016** [comment]
@EgorDuplensky ,Updated.

**Review by EgorDuplensky — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:1626]: I guess the comments should be updated.
Could you please also check all the corresponding comments in terms of relevance.

Comment (luweizhou2016) [reply src/plugins/intel_cpu/src/nodes/conv.cpp]: Updated. Thx!

**Comment by luweizhou2016** [comment]
The current code inclue:
1. Enable per-tensor zp on AVX512_VNNI and AVX512_AMX ISA to use BRGCONV kernel.
2. Merge the @luo-cheng2021  fix PR https://github.com/openvinotoolkit/openvino/pull/12877
3. Merge optimization of convolution attribute PR to lower down the load network time latency.https://github.com/openvinotoolkit/openvino/pull/12898
4. Only enable conv/group conv S8 input on AMX. Non-AMX same with 2022.1 LPT input precision config.

**Comment by maxnick** [comment]
@EgorDuplensky , could you please take a look once again since the PR incorporated a couple of other ones.

File: src/plugins/intel_cpu/src/graph_optimizer.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/graph_optimizer.cpp:640]: I propose we hide all this inside convolution node.
In method like:
```c++
initializeInputZeroPoints(const uint8_t* zpData, const size_t zpSize)
```

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:959]: Please avoid links and leave only 8558 as a ticket number

File: src/plugins/intel_cpu/src/plugin.cpp
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/plugin.cpp:524]: Shouldn't we do it in scope of separate PR and benchmark it separately?
As far as I understand this PR is mostly about amx issues.
But here we are changing non-amx behavior as well (disabling s8 support)

Comment (luweizhou2016) [reply src/plugins/intel_cpu/src/plugin.cpp:524]: Yes. Since we have regression on VNNI when switching back to int8 input type. Switching back to U8 on VNNI needs to handle zero point and test. So that is why I add this part into this. Will update this PR description. In fact, we can use this PR to trigger related test for easiness. Thx!

**Comment by luweizhou2016** [comment]
@EgorDuplensky , Review comments applied. Pls help to review. Thx!

File: src/plugins/intel_cpu/src/plugin.cpp
Comment (EgorDuplensky) [reply src/plugins/intel_cpu/src/plugin.cpp:524]: I mean yes, I understand the reason, but in general, I think it is much more convenient just to have a separate commit for this which we can refer to in case of issues.
Just to not mix all the fixes in scope of one commit / PR
@maxnick @dmitry-gorokhov What do you think?

Comment (EgorDuplensky) [reply src/plugins/intel_cpu/src/plugin.cpp:524]: discussed, it seems we have no choice and have to do it in scope of single PR
So the comment can be ignored

**Review by EgorDuplensky — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/low_precision_transformations/fake_quantize_with_dq_not_optimal_transformation.cpp
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/low_precision_transformations/fake_quantize_with_dq_not_optimal_transformation.cpp:45]: Commented code detected. Please either uncomment the code frame, or make a ticket and disable the test via the common mechanism (e.g. https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp#L178, though it is commented the mechanics is the same).

**Review by maxnick — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/plugin.cpp
Comment (luweizhou2016) [reply src/plugins/intel_cpu/src/plugin.cpp]: ConvolutionBackpropData has supported the I8/U8 input precision in 2022.1 release. And why we disabled I8 input for avx512-vnni is that onednn I8 input compensation implement (2022.2 introduced) perf is lower than U8 OV legacy input zero point(2022.1).  As for  ConvolutionBackpropData, First we don't have legacy input zero point mechanism in OV and we should keep same config with 2022.1 to avoid regression. So I think we should keep support I8 input precision for ConvolutionBackpropData as 2022.1. Do you think it is okay?

**Comment by luweizhou2016** [comment]
Thx @EgorDuplensky @maxnick . Applied reveiw comments and pls review @maxnick .

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:3]:
```suggestion
input zero-point were introduced into OpenVINO before mature archetecture is developed in onednn. Now, onednn has finalized the post-ops/zero-points
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:5]:
```suggestion
optimization mechanism/API and has some diverges with OV mechanism. To benefit perf improvement from onednn for a long term, no further investment will be applied onto
legacy post-ops/zero-point. So current code will have to support both legacy post-ops/zero-point and onednn post-ops/zero point mechanism.
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:17]:
```suggestion
**dwconv is not fused on CPU above avx2. So no need to consider.**
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:24]:
```suggestion
OpenVINO legacy input zero point can support per-channel input tensor zero point. The pre-calculated output compensation will
be passed into onednn forked kernel.
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:30]:
```suggestion
Fused legacy per-channel input zero point can not be supported on amx cpu platforms and will fall back on vnni. The perf would be greatly lower than amx.
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:34]:
```suggestion
On AMX platform, even if the conv fuses binary post-ops, conv node will try to use brgconv no matter whether the app enforces brgconv.
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:36]:
```suggestion
On AVX-512 platform, when conv fuses binary post-ops, the conv node will try to use brgconv only when the app enforce it via rtinfo.
```

File: src/plugins/intel_cpu/src/docs/convPostOps.md
Comment (maxnick) [inline src/plugins/intel_cpu/src/docs/convPostOps.md:64]:
```suggestion
**When there is no legacy post ops or legacy zero point, only need attr[0].**
```

**Review by maxnick — APPROVED** [review]

**Comment by luweizhou2016** [comment]
Thx! Applied review comments.

**Comment by maxnick** [comment]
@dmitry-gorokhov, could you please take a look?

File: src/plugins/intel_cpu/src/nodes/conv.h
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/conv.h:56]: I would suggest to rename: `stockInputZeroPoints` -> `inputZeroPoints`

File: src/plugins/intel_cpu/src/nodes/conv.h
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/conv.h:157]: Seems like this variable is not used.

File: src/plugins/intel_cpu/src/nodes/conv.cpp
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:932]: In general current impl looks reasonable. At the same time it still can be improved. Examples:

- On avx2 systems for Convolution with per-tensor zp or fused quantization we still create 2 versions of attrs. This doesn't make any sense since Brgemm Convs are not supported on AVX2 - only legacy attrs required.
- On avx512 systems for Convolution with per-tensor zp we still create 2 versions of attrs. Obviously this case is supported via Brgemm Convs and legacy attrs are not required.

Both cases should be resolved as part of first inference latency optimization activity. Doesn't block this PR.

Comment (luweizhou2016) [reply src/plugins/intel_cpu/src/nodes/conv.cpp:932]: - On avx2, shouldTryBrgconv should be false except user application enforce this via RTinfo. So most of the avx2 should only use legacy attribute and would not try brgconv attribute.
- On avx512, Yes. We would have per-tensor zp targeting brgconv kernel and  legacy output compensation targeting JIT kernel.  Current brgconv in ONEDNN has unimplement early return with some heuristic code. So have to use legacy attribute as backup if brgconv can't hit. For long-term, maybe we can remove it if onednn remove these limitation. Thx!

**Comment by luweizhou2016** [comment]
Updated. Thx! @dmitry-gorokhov

**Review by dmitry-gorokhov — APPROVED** [review]
Can be merged after internal validation

**Pull request merged**
